Fix opus_gnome dataset card

gojiteji opened the pull request:
I fixed a issue #4805.

I changed `"gnome"` to `"opus_gnome"` in[ README.md](https://github.com/huggingface/datasets/tree/main/datasets/opus_gnome#dataset-summary).

Fix #4805

HuggingFaceDocBuilderDev commented:
The docs for this PR live [here](https://moon-ci-docs.huggingface.co/docs/datasets/pr_4806). All of your documentation changes will be reflected on that endpoint.

albertvillanova commented:
@gojiteji why have you closed this PR and created an identical one?
- #4807

gojiteji commented:
@albertvillanova 
I forgot to follow "How to create a Pull" in CONTRIBUTING.md in this branch.

albertvillanova commented:
Both are identical. And you can push additional commits to this branch.

gojiteji commented:
I see. Thank you for your comment.

albertvillanova commented:
Anyway, @gojiteji thanks for your contribution and this fixing.

gojiteji commented:
> the errors that appear have nothing to do with your contribution, but with these additional quality requirements.

Is there anything I should do?

albertvillanova commented:
If you would like to address them as well in this PR, it would be awesome: https://github.com/huggingface/datasets/runs/7741104780?check_suite_focus=true

albertvillanova commented:
This is the error message:
```
E           ValueError: The following issues have been found in the dataset cards:
E           README Validation:
E           The following issues were found for the README at `/home/runner/work/datasets/datasets/datasets/opus_gnome/README.md`:
E           -	No first-level heading starting with `Dataset Card for` found in README. Skipping further validation for this README.
E           The following issues have been found in the dataset cards:
E           YAML tags:
E           Could not validate the metadata, found the following errors:
E           * field 'language':
E           	['ara', 'cat', 'foo', 'gr', 'nqo', 'tmp'] are not registered tags for 'language', reference at https://github.com/huggingface/datasets/tree/main/src/datasets/utils/resources/languages.json
```

albertvillanova commented:
In principle there are 2 errors:

The first one says, the title of the README does not start with `Dataset Card for`:
- The README title is: `# Dataset Card Creation Guide`
- According to the [template here](https://github.com/huggingface/datasets/blob/main/templates/README.md), it should be: `# Dataset Card for [Dataset Name]`

albertvillanova commented:
In relation with the languages:
- you should check whether the language codes are properly spelled
- and if so, adding them to our `languages.json` file, so that they are properly validated

gojiteji commented:
Thank you for the detailed information. I'm checking it now.

gojiteji commented:
How can I know whether these `'ara', 'cat', 'foo', 'gr', 'nqo', 'tmp'` are misspelled language codes or lack of `languages.json`?

albertvillanova commented:
```
E           ValueError: The following issues have been found in the dataset cards:
E           README Validation:
E           The following issues were found for the README at `/home/runner/work/datasets/datasets/datasets/opus_gnome/README.md`:
E           -	Expected some content in section `Data Instances` but it is empty.
E           -	Expected some content in section `Data Fields` but it is empty.
E           -	Expected some content in section `Data Splits` but it is empty.
```

gojiteji commented:
I added `ara`, `cat`, `gr`, and `nqo` to  `languages.json` and removed `foo` and `tmp` from `README.md`.
I also write  Data Instances, Data Fields, and Data Splits in `README.md`.

albertvillanova commented:
Thanks for your investigation and fixes to the dataset card structure! I'm just making some suggestions before merging this PR: see below.

albertvillanova commented on datasets/opus_gnome/README.md:
```diff
@@ -41,7 +41,6 @@ language:
 - fa
 - fi
 - fo
-- foo
```
Thanks for removing this tag. I searched for it but I was not able to find any reference to it. Looking at the samples in opus_gnome, it seems just English. I'm wondering where this `foo` tag comes from...

If we cannot find any external reference to it, I agree it is better to remove it.

albertvillanova commented on datasets/opus_gnome/README.md:
```diff
@@ -130,7 +129,6 @@ language:
 - th
 - tk
 - tl
-- tmp
```
This is a deprecated language tag: it should be replaced by `tyj`.

However, `tyj` should be added to our `languages.json`:
```
"tyj": "Tai Do; Tai Yo",
```

See reference here: https://iso639-3.sil.org/request/2015-019

gojiteji commented:
Should I create PR for `config.json` to add ` ara cat gr nqo` first?
I think I can pass this failing after that.

Or removing `ara, cat, gr, nqo, foo, tmp` from `README.md`.

albertvillanova commented on src/datasets/utils/resources/languages.json:
```diff
@@ -58,6 +58,7 @@
     "ar-TD": "Arabic (Chad)",
     "ar-TN": "Arabic (Tunisia)",
     "ar-YE": "Arabic (Yemen)",
+    "ara":"Arabic",
```
When there exist several codes (ISO 639 1, 2, 3) for the same language, we just use the lower one. For Arabic, we use `ar`.

As `ar` already appears in the README, I would remove `ara` from the README and would not add it to `languages.json`

```suggestion
```

albertvillanova commented on src/datasets/utils/resources/languages.json:
```diff
@@ -139,6 +140,7 @@
     "byn": "Blin; Bilin",
     "bzd": "Bribri",
     "ca": "Catalan",
+    "cat": "Catalan; Valencian",
```
The same as above: for "Catalan; Valencian" we use `ca` instead. I would remove `cat` from README and from here.
```suggestion
```

albertvillanova commented on src/datasets/utils/resources/languages.json:
```diff
@@ -523,6 +525,7 @@
     "gom": "Goan Konkani",
     "gor": "Gorontalo",
     "got": "Gothic",
+    "GR": "Greek",
```
Than language code for Greek is `el`. I would remove `gr` from README and from here.
```suggestion
```

albertvillanova commented on src/datasets/utils/resources/languages.json:
```diff
@@ -861,6 +864,7 @@
     "not": "Nomatsiguenga",
     "nov": "Novial",
     "npi": "Nepali (individual language)",
+    "nqo": "N'Ko",
```
Well done!

gojiteji replied:
Thank you for your suggestion very much.
And sorry for a lot of CI tests.

albertvillanova reviewed (changes requested):
My suggestions for the language codes.

albertvillanova commented:
Once you address these issues, all the CI tests will pass.

gojiteji reviewed (commented):
Thank you for your request, and sorry for the many CI tests runs.

albertvillanova commented:
Once the remaining changes are addressed (see unresolved above), we will be able to merge this:
- [ ] Remove "ara" from README
- [ ] Remove "cat" from README
- [ ] Remove "gr" from README
- [ ] Replace "tmp" with "tyj" in README
- [ ] Add "tyj" to `languages.json`:
  ```
  "tyj": "Tai Do; Tai Yo",

gojiteji commented:
I did the five changes.

albertvillanova reviewed (approved):
Thanks, @gojiteji for the original fix and all the subsequent improvements to the dataset documentation card.
